sgd/src/main/java/RunSGD.java

```diff
@@ -0,0 +1,244 @@
+import org.qcri.rheem.api.DataQuantaBuilder;
+import org.qcri.rheem.api.JavaPlanBuilder;
+import org.qcri.rheem.basic.data.Tuple2;
+import org.qcri.rheem.core.api.RheemContext;
+import org.qcri.rheem.core.function.ExecutionContext;
+import org.qcri.rheem.core.function.FunctionDescriptor;
+import org.qcri.rheem.core.util.RheemCollections;
+import org.qcri.rheem.core.util.Tuple;
+import org.qcri.rheem.java.Java;
+import org.qcri.rheem.spark.Spark;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Created by zoi on 8/27/16.
+ */
+public class RunSGD {
+
+
+    static String relativePath = "sgd/src/main/resources/adult.zeros";
+    static int datasetSize  = 100827;
+    static int features = 123;
+    static boolean sparse = true;
+    static boolean binary = true; //this applies only for SVM, to transform the 0 to -1 during the transform
+
+    //these are for SGD/mini run to convergence, put < 1 for batch GD
+    static int sampleSize = 6;
+    static double accuracy = 0.001;
+    static int max_iterations = 1000;
+
+
+    public static void main (String... args) throws MalformedURLException {
+
+        //Usage: <data_file> <#features> <sparse> <binary>
+        if (args.length > 0) {
+            relativePath = args[0];
+            datasetSize = Integer.parseInt(args[1]);
```

Comment:
It's going to be more complicated the plan because I already have a broadcast for the weights. Let's keep it simple